src/System.IO.FileSystem/src/System/IO/Win32FileSystemEnumerable.cs

```diff
@@ -130,25 +136,25 @@ internal Win32FileSystemEnumerableIterator(String path, String originalUserPath,
             else
             {
                 _resultHandler = resultHandler;
-                this._searchOption = searchOption;
+                _searchOption = searchOption;
 
                 _fullPath = Path.GetFullPath(path);
-                String fullSearchString = GetFullSearchString(_fullPath, normalizedSearchPattern);
+                string fullSearchString = GetFullSearchString(_fullPath, normalizedSearchPattern);
                 _normalizedSearchPath = Path.GetDirectoryName(fullSearchString);
 
                 // normalize search criteria
                 _searchCriteria = GetNormalizedSearchCriteria(fullSearchString, _normalizedSearchPath);
 
                 // fix up user path
-                String searchPatternDirName = Path.GetDirectoryName(normalizedSearchPattern);
-                String userPathTemp = originalUserPath;
+                string searchPatternDirName = Path.GetDirectoryName(normalizedSearchPattern);
+                string userPathTemp = originalUserPath;
                 if (searchPatternDirName != null && searchPatternDirName.Length != 0)
                 {
                     userPathTemp = Path.Combine(userPathTemp, searchPatternDirName);
                 }
-                this._userPath = userPathTemp;
+                _userPath = userPathTemp;
```

Comment:
Again just personal preference, but if it were me I'd have written the last six lines instead as:
```C#
_userPath = string.IsNullOrEmpty(searchPatternDirName) ?
    originalUserPath :
    Path.Combine(originalUserPath, searchPatternDirName);
```